tokio/src/runtime/task/id.rs

```diff
@@ -74,11 +74,17 @@ impl fmt::Display for Id {
 
 impl Id {
     pub(crate) fn next() -> Self {
-        use crate::loom::sync::atomic::{Ordering::Relaxed, StaticAtomicU64};
+        #[cfg(loom)]
+        return Self(context::next_task_id().unwrap());
```

Comment:
I think what you can do is use loom's lazy_static wrapper, which you can read about at https://docs.rs/loom/latest/loom/macro.lazy_static.html

This avoids a problem where tasks created in different threads get the same task id, since thread locals have a separate counter per thread.

By using the lazy_static wrapper, you should be able to put a `loom::sync::atomic::AtomicU64` in a global.

Reply:
I have no tests on it yet, but as far as I know, the lazy_static has a little performance loss, do you think it is suitable for generating global ID?

Reply:
You should only use lazy_static during loom tests, so this does not impact performance when not using loom.

Reply:
I get you pointer, I will try this way.

Reply:
But when I try this way,
```
  #[cfg(loom)]
        {
            crate::loom::lazy_static! {
                static ref NEXT_ID: StaticAtomicU64 = StaticAtomicU64::new(1);
            }            
            Self(NEXT_ID.fetch_add(1, Relaxed)) 
        }
```

 this following error occurs:

```
error[E0433]: failed to resolve: could not find `lazy_static` in `loom`
  --> tokio/src/runtime/task/id.rs:84:26
   |
84 |             crate::loom::lazy_static! {
   |                          ^^^^^^^^^^^ could not find `lazy_static` in `loom`
```

Reply:
Oh may be we should use `[cfg(not(all(test, loom)))]` and `#[cfg(all(test, loom))]` instead.
`

Reply:
I  have use `lazy_static` in loom instead.